flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/operations/SqlToOperationConverter.java

```diff
@@ -612,10 +668,14 @@ private TableSchema createTableSchema(SqlCreateTable sqlCreateTable) {
 			if (node instanceof SqlTableColumn) {
 				SqlTableColumn column = (SqlTableColumn) node;
 				RelDataType relType = column.getType()
-					.deriveType(validator, column.getType().getNullable());
+					.deriveType(validator);
+				RelDataType colType = validator.getTypeFactory()
+						.createTypeWithNullability(
+								relType,
+								sqlCreateTable.isColumnNullable(column));
```

Comment:
I think we should modify the nullability in the `schemaBuilder.primaryKey(...)`. 
Otherwise, the `TableSchema` is not a error-prone structure. 
Otherwise, we have to do the same thing for descriptor API in the future.